core/src/main/java/org/web3j/ens/NameHash.java

```diff
@@ -69,4 +71,36 @@ public static String normalise(String ensName) {
             throw new EnsResolutionException("Invalid ENS name provided: " + ensName);
         }
     }
+
+    public static byte[] toUtf8Bytes(String string) {
+        if (string == null || string.isEmpty()) {
+            return null;
+        }
+        return string.getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Encode Dns name. Reference implementation
+     * https://github.com/ethers-io/ethers.js/blob/fc1e006575d59792fa97b4efb9ea2f8cca1944cf/packages/hash/src.ts/namehash.ts#L49
+     *
+     * @param name Dns name
+     * @return Encoded name in Hex format.
+     * @throws IOException
+     */
+    public static String dnsEncode(String name) throws IOException {
+        String[] parts = name.split("\\.");
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        for (String part : parts) {
+            byte[] bytes = toUtf8Bytes("_" + normalise(part));
+            if (bytes == null) {
+                break;
+            }
+            bytes[0] = (byte) (bytes.length - 1);
+
+            outputStream.write(bytes);
+        }
+
+        return Numeric.toHexString(outputStream.toByteArray()) + "00";
```

Comment:
just to know, why is necessary to append 00?

Reply:
I think it is used as a marker of the end.